kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -283,6 +296,7 @@ public class TableConfig<T> {
               addConfig(this, ICEBERG_COMPAT_V2_ENABLED);
               addConfig(this, COLUMN_MAPPING_MAX_COLUMN_ID);
               addConfig(this, DATA_SKIPPING_NUM_INDEXED_COLS);
+              addConfig(this, ICEBERG_WRITER_COMPAT_V1_ENABLED);
```

Comment:
nit: maybe keep this together with `ICEBERG_COMPAT_V2_ENABLED`

Reply:
Including this in https://github.com/delta-io/delta/pull/4291 :)